src/ripple/app/ledger/LedgerHistory.cpp

```diff
@@ -158,26 +159,114 @@ log_metadata_difference(Ledger::pointer builtLedger, Ledger::pointer validLedger
     builtLedger->getTransactionMeta(tx, builtMetaData);
     assert(validMetaData != nullptr || builtMetaData != nullptr);
+
     if (validMetaData != nullptr && builtMetaData != nullptr)
     {
-        WriteLog (lsERROR, LedgerMaster) <<
-            "MISMATCH tx differ in metadata only " << tx <<
-            " : built metadata is " << builtMetaData->getJson(0) <<
-            " : valid metadata is " << validMetaData->getJson(0);
+        auto const& validNodes = validMetaData->getNodes ();
+        auto const& builtNodes = builtMetaData->getNodes ();
+
+        bool result_diff =
+            validMetaData->getResultTER () != builtMetaData->getResultTER ();
+
+        bool const index_diff =
+            validMetaData->getIndex() != builtMetaData->getIndex ();
+
+        bool const nodes_diff = validNodes != builtNodes;
+
+        if (!result_diff && !index_diff && !nodes_diff)
+        {
+            WriteLog (lsERROR, LedgerMaster) << "MISMATCH on TX " << tx <<
+                ": No apparent mismatches detected!";
+            return;
+        }
+
+        if (!nodes_diff)
+        {
+            if (result_diff && index_diff)
+            {
+                WriteLog (lsERROR, LedgerMaster) << "MISMATCH on TX " << tx <<
+                    ": Different result and index!";
+                WriteLog (lsERROR, LedgerMaster) << " Built:" <<
+                    " Result: " << builtMetaData->getResult () <<
+                    " Index: " << builtMetaData->getIndex ();
+                WriteLog (lsERROR, LedgerMaster) << " Valid:" <<
+                    " Result: " << validMetaData->getResult () <<
+                    " Index: " << validMetaData->getIndex ();
+            }
+            else if (result_diff)
+            {
+                WriteLog (lsERROR, LedgerMaster) << "MISMATCH on TX " << tx <<
+                    ": Different result!";
+                WriteLog (lsERROR, LedgerMaster) << " Built:" <<
+                    " Result: " << builtMetaData->getResult ();
+                WriteLog (lsERROR, LedgerMaster) << " Valid:" <<
+                    " Index: " << validMetaData->getResult ();
+            }
+            else if (index_diff)
+            {
+                WriteLog (lsERROR, LedgerMaster) << "MISMATCH on TX " << tx <<
+                    ": Different index!";
+                WriteLog (lsERROR, LedgerMaster) << " Built:" <<
+                    " Index: " << builtMetaData->getIndex ();
+                WriteLog (lsERROR, LedgerMaster) << " Valid:" <<
+                    " Index: " << validMetaData->getIndex ();
+            }
+        }
+        else
+        {
+            if (result_diff && index_diff)
+            {
+                WriteLog (lsERROR, LedgerMaster) << "MISMATCH on TX " << tx <<
+                    ": Different result, index and nodes!";
+                WriteLog (lsERROR, LedgerMaster) << " Built:\n" <<
+                    builtMetaData->getIndex ();
```

Comment:
`WriteLog` formatting is a bit different for this case compared to the others.  For consistency consider adding `" Index: " << xxxxxMetaData...` to lines 222 and 224.